dev/validate_bom__test.py

```diff
@@ -578,6 +575,13 @@ def wait_on_service(self, service_name, port=None, timeout=None):
     Returns:
       The ForwardedPort entry for this service.
     """
+    if len(self.__outbound_service_configs) > 0 and service_name not in self.__outbound_service_configs:
```

Comment:
This is equivalent to:
```
if self.__outbound_service_configs:
  if service_name in self.__outbound_service_configs:
     self.__wait_on_service_using_endpoint(service_name)
  return True

try:
 ...
```

But that doesnt look right, does it?
If it is in the outbound_service_configs then use the endpoint, otherwise dont.
Or do you really mean to not wait on the other services? That's ok if intentional,
in which case the simpler above makes it more clear. Throw a comment in as to
why you always return True if a non-empty outbound_service_cofnigs.

Also, didnt we change "outbound" to something else?

Reply:
I will add a comment indicating that we do not (actually cannot) wait for the other services, since we are expected to communicate with services via a url (instead of port forwarding).

Would it be clearer if we take an explicit --no-port-forward=true or --external-services=true in order to achieve our flow?

Reply:
I assume this is all or none. I think maybe the options specify whether the services are on public addresses or private. Private addresses require port forwarding. Or more precisely you need to know the network the services are on and whether they are on your network or not. A private network that you are on would not require port forwarding either.

If you go this general route, then the controller would need to figure out if it is on the same network or not. I think that's kind of what it does now to port forward. It looks up the metadata for the machine it wants to connect to then tries the private network, then the public network. At least for VMs. Kubernetes might be one way. I wonder if you can tell if you are on the same network by looking at the IP address you want to connect to and determining if it is public or private and, if private, see if you have an IP address on that same private network.

Is that too much? For now you can add some kind of option indicating whether to port forward or not. The default for port forward is true so make no-port-forward set port_forward to false.

Actually, lookup up an example for you I came across this:
    parser.add_argument(
        '--deploy_google_use_external_ip',
        dest='deploy_google_use_internal_ip', action='store_false',
        help='DEPRECATED: Use --deploy_google_use_internal_ip=false')

Maybe you can just use that flag. Set --deploy_google_use_internal_ip=false and then use that in the test module to connect with the external address. Does that satisfy your need?